File: employee-importance.py
from typing import List, int


# Definition for Employee.
class Employee:
    def __init__(self, id: int, importance: int, subordinates: List[int]):
        self.id = id
        self.importance = importance
        self.subordinates = subordinates


class Solution:
    def getImportance(self, employees: List[Employee], id: int) -> int:

        def dfs(root):
            ans = root.importance
            for i in root.subordinates:
                ans += dfs(mpp[i])
            return ans

        mpp = dict()
        req = None

        for i in employees:
            mpp[i.id] = i
            if i.id == id:
                req = i
        return dfs(req)

File: employee-importance.cpp
/*
// Definition for Employee.
class Employee {
public:
    int id;
    int importance;
    vector<int> subordinates;
};
*/

class Solution {
public:
    int dfs(Employee* emp,unordered_map <int, Employee*> mpp) {
        int ans = emp->importance;
        for (int id:emp->subordinates){
            ans += dfs(mpp[id],mpp);
        }
        return ans;

    }

    int getImportance(vector<Employee*> employees, int id) {
        Employee* req_employee = NULL;
        unordered_map <int, Employee*> mpp;
        for (auto i:employees){
            mpp.insert({i->id,i});
             if (i->id == id) {
                req_employee = i;
            }
        }
        // cout<<req_employee->id;

        return dfs(req_employee,mpp);

    };
};
